CONTRIBUTING.md

```diff
@@ -35,9 +38,37 @@ To contribute to this repository:
 
           $ git push -u origin my-new-branch
 
-Finally, go to the web page of your fork of the `everware` repo,
+Finally, go to the web page of your fork of the `everware` repository,
 and click 'Pull request' to send your changes to the maintainers for
-review. This will send an email to the committers.
+review. This will send an email to the committers. A valid PR includes
+the following items:
+
+* a description of the problem you are solving or reference to existing issue
+* a discussion why you made the choices you made
+* contains the minimal set of changes needed to solve the problem (do
+  not mix several issues, rather create separate PRs)
+* unit and frontend tests for your changes
+* the code should follow the style of the project
+* the code should contain documentation in the form of docstrings
+  and comments
+* no commits merging master into your branch, instead rebase your branch
+
+These are guidelines and not hard rules, so if you feel one of them does not
+apply, write a short sentence in the PR's description explaining why you
+skipped it.
+
+We encourage you to start your PR as soon as possible to get feedback early.
```

Comment:
the following 3 paragraphs are like soft rules, but the one about 'not merging own PRs' is more hard. maybe put it before softer ones?